android/src/main/java/com/clevertap/react/CleverTapApplication.kt

```diff
@@ -0,0 +1,16 @@
+package com.clevertap.react
+
+import com.clevertap.android.sdk.Application
+import com.clevertap.android.sdk.CleverTapAPI
+
+open class CleverTapApplication : Application() {
+    override fun onCreate() {
+        super.onCreate()
+        // Create SDK instance and attach the proxy listener when the app is created. This
+        // handles events which are triggered while the CleverTap react module is not yet
+        // initialized
+        CleverTapAPI.getDefaultInstance(this)?.let {
```

Comment:
The previous fix for NOTIFICATION_CLICK event was only done in this class, that's why I've only included it here.  It would be easy to integrate this in a custom Application class. We have a few options:

* Add the same initialization code in the documentation and let people know that they should include this in their ```Application#onCreate```. :
```kotlin
CleverTapAPI.getDefaultInstance(this)?.let {
    CleverTapListenerProxy.attachToInstance(it)
}
```

* We can add a more purposefully named util method that does the same, for example:

```kotlin
object CleverTapRnAPI {
    fun initReactNativeIntegration(context: Context) {
        CleverTapAPI.getDefaultInstance(context)?.let {
            CleverTapListenerProxy.attachToInstance(it)
        }
    }
}
```

and add to the docs that people should add this to their ```onCreate```:
```kotlin
CleverTapRnAPI.initReactNativeIntegration(this)
```

Reply:
This will ease integration but will be a breaking change and needs to be documented


Not sure but a client or 2 might be having an integration where they don't use the `CleverTapApplication` class at all (they might not be using `NotificationClicked` event from killed state). In such a case other events would have worked fine (inbox, inapp etc. related) for them as these events weren't tied to the application class

Reply:
If they don't include this code in their Application class, the behavior of the events would remain the same as it was previously. This line only registers listening to the events as early as possible, but the listeners are also registered in CleverTapModuleImpl#initCtInstance to cover the case when they might have not been registered in the Application class.

Reply:
Makes sense

Reply:
Also ClevertapListenerProxy is not public, can this be changed?